### PR TITLE
Create terraform variable for secret length

### PR DESCRIPTION
**Pull request by dkatzz**
https://github.com/civiform/civiform/issues/6506

We want to create a terraform variable for the random password length, so certain deployments can update it before they have users.

We'll need to update this before the play framework migration (https://www.playframework.com/documentation/2.9.x/Migration29#Application-Secret-enforces-a-minimum-length), and it will be easier to change the length for the deployments without many users now.

### CODE REVIEW

File: cloud/aws/templates/aws_oidc/variables.tf [hunk 1]
```diff
@@ -472,3 +472,9 @@ variable "monitoring_stack_enabled" {
   description = "If true, Prometheus and Grafana instances are created."
   default     = true
 }
+
+variable "random_password_length" {
+  type        = number
+  description = "Length of the random generated password to use for app_secret_key"
+  default     = 16
```
Comment (gwendolyngoetz) [inline cloud/aws/templates/aws_oidc/variables.tf:479]: The default should be at least 32 or Play won't start once we merge in v2.9. I'd suggest making it 64 so we're set if we ever need to increate the signature algorithm from HS256 to HS512.

File: cloud/aws/templates/aws_oidc/secrets.tf [hunk 2]
```diff
@@ -79,7 +79,7 @@ resource "aws_secretsmanager_secret_version" "postgres_password_secret_version"
 
 # Create a random generated password to use for app_secret_key.
 resource "random_password" "app_secret_key" {
-  length           = 16
+  length           = var.random_password_length
```
Comment (gwendolyngoetz) [inline cloud/aws/templates/aws_oidc/secrets.tf:82]: Is there a way to error if this is smaller than the default?

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

File: cloud/aws/templates/aws_oidc/variables.tf (hunk 1 above)
Comment (dkatzz) [reply cloud/aws/templates/aws_oidc/variables.tf:479]: I just don't want to break any existing deployments with the session stuff, but I think the plan is to change this default to higher once we know the implications and then do the play migration.

File: cloud/aws/templates/aws_oidc/secrets.tf (hunk 2 above)
Comment (dkatzz) [reply cloud/aws/templates/aws_oidc/secrets.tf:82]: Yeah! I can add a validation, but we can do that once we update this default to the higher one.

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**